### PR TITLE
trying to add sourcemaps support

### PR DESCRIPTION
**Pull request by shelldandy**
From #13

### CODE REVIEW

**Comment by avaly** [comment]
Thanks for the contribution! Is this change testable as well? Can we assert the a source map has been generated in the unit tests?

**Comment by shelldandy** [comment]
Im not sure how to programatically test it out.

I just have a gulp task that handles scss and does a couple extra things to it like autoprefixer and so on.

Before this, when opening Chrome Developer Tools the sourcemaps would point to a wrong file or location number, after this they point to the correct source

File: index.js [hunk 1]
```diff
@@ -2,9 +2,14 @@
 var PluginError = require('plugin-error');
 var through = require('through2');
 var gcmq = require('group-css-media-queries');
+var applySourceMap = require('vinyl-sourcemaps-apply');
 
 module.exports = function () {
 	return through.obj(function (file, enc, cb) {
+		// generate source maps if plugin source-map present
+		if (file.sourceMap) {
+			options.makeSourceMaps = true;
```
Comment (avaly) [inline index.js]: `options` is not defined any where and is not being used anywhere else.

File: index.js [hunk 2]
```diff
@@ -17,6 +22,10 @@ module.exports = function () {
 
 		try {
 			file.contents = new Buffer(gcmq(file.contents.toString()));
+			// apply source map to the chain
+			if (file.sourceMap) {
+				applySourceMap(file, result.map);
```
Comment (avaly) [inline index.js:27]: `result` is undefined and not used anywhere else.

**Review by avaly — CHANGES_REQUESTED** [review]
Thanks for your time, but this does not seem to be valid code. Also, I want to have proper unit tests for any changes made to the codebase. Please try to figure out how to get the code valid and tested. Thanks!

File: index.js (hunk 2 above)
Comment (shelldandy) [reply index.js:27]: Maybe adding a linter might help (?)

Do you want me to add one?

Comment (avaly) [reply index.js:27]: Feel free to add `eslint` in another PR.

**Pull request closed without merging**